oci/table_oci_functions_function.go

```diff
@@ -0,0 +1,354 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	"github.com/oracle/oci-go-sdk/v44/common"
+	"github.com/oracle/oci-go-sdk/v44/functions"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableFunctionsFunction(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_functions_function",
+		Description: "OCI Functions Function",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getFunction,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listFunctionsApplications,
+			Hydrate:       listFunctions,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "lifecycle_state",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "The display name of the function.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the function.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "application_id",
+				Description: "The OCID of the application the function belongs to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ApplicationId"),
```

Comment:
Do we need this transform field?

Reply:
The data is not being displayed in the table otherwise. So I added it.